### PR TITLE
SWTBot Fix 3: Improved Workspace Synchronization 

### PR DESCRIPTION
**Pull request by ghost**
- Fixed issue with creating new projects not being correctly
synchronized with workspace locking.
- Removed SaveJob since it is not needed anymore.
- Refactored workspace locked execution in editing domain not to use
code from Command Stack anymore. The command stack is checking for a
save trigger to be set, and tries to issue a save. Deadlocks may happen
or race conditions when not correctly setting the save trigger.
- Improved logging in command stack.
- Improved save trigger handling in command stack.

---
Task #460: Try to improve SWTBot stability

### CODE REVIEW

File: de.dlr.sc.virsat.project/src/de/dlr/sc/virsat/project/editingDomain/VirSatWorkspaceCommandStack.java [hunk 1]
```diff
@@ -224,20 +227,22 @@ public void setEditingDomain(VirSatTransactionalEditingDomain editingDomain) {
...
-	private void checkTriggerSaveAll() {
+	private synchronized void checkTriggerSaveAll() {
+		Activator.getDefault().getLog().log(new Status(Status.INFO, Activator.getPluginId(), "VirSatWorkspaceCommandStack: Check for save trigger"));
 		if (triggerSave) {
-			Activator.getDefault().getLog().log(new Status(IStatus.INFO, Activator.getPluginId(), "Executed command triggered save all on editing domain"));
-			VirSatSaveJob saveJob = new VirSatSaveJob(editingDomain, true);
-			saveJob.scheduleIfNoSaveJobPending();
-			triggerSave = false;
+			Activator.getDefault().getLog().log(new Status(IStatus.INFO, Activator.getPluginId(), "VirSatWorkspaceCommandStack: Save tiggered Scheduling Save Job."));
+			// This boolean has to be reset first, otherwise the save all will be executed in
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.project/src/de/dlr/sc/virsat/project/editingDomain/VirSatWorkspaceCommandStack.java]: Which boolean are we talking about in this comment?

**Review by SaMuellerDLR — CHANGES_REQUESTED** [review]

File: de.dlr.sc.virsat.project/src/de/dlr/sc/virsat/project/editingDomain/VirSatWorkspaceCommandStack.java (hunk 1 above)
Comment (PhilMFischer) [reply de.dlr.sc.virsat.project/src/de/dlr/sc/virsat/project/editingDomain/VirSatWorkspaceCommandStack.java]: Has been a deprecated comment. Updated it.

Comment (SaMuellerDLR) [reply de.dlr.sc.virsat.project/src/de/dlr/sc/virsat/project/editingDomain/VirSatWorkspaceCommandStack.java]: @PhilMFischer I dont see any new commits.

**Review by SaMuellerDLR — APPROVED** [review]
Approved!

**Review by franzTobiasDLR — APPROVED** [review]

**Pull request merged**